perf_test/batched/KokkosBatched_Test_BlockTridiagDirect.cpp

```diff
@@ -0,0 +1,595 @@
+                                                                              bv.extent(2),
+                                                                              bv.extent(3),
+                                                                              vector_length/internal_vector_length);
+    
+
+    /// double copy of A
+    Kokkos::View<value_type******,Kokkos::LayoutRight,exec_space> Acopy("Acopy",
+                                                                        As.extent(0),
+                                                                        As.extent(1),
+                                                                        As.extent(2),
+                                                                        As.extent(3),
+                                                                        As.extent(4),
+                                                                        As.extent(5));
+
+    Kokkos::View<value_type*****,Kokkos::LayoutRight,exec_space> rs("rs",
+                                                                    bs.extent(0), 
+                                                                    bs.extent(1), 
+                                                                    bs.extent(2), 
+                                                                    bs.extent(3), 
+                                                                    bs.extent(4));
+
+    auto AA = Ai;
+    auto bb = bi;
+    auto xx = xi;
+
+    // auto AA = As;
+    // auto bb = bs;
+    // auto xx = xs;
+
+    ///
+    /// set identity
+    ///
+    if (0) {
+#if defined(KOKKOS_ENABLE_CUDA) && defined(KOKKOSBATCHED_PROFILE)
+      cudaProfilerStart();
+#endif
+      timer.reset();
+      using policy_type = Kokkos::TeamPolicy<exec_space>;
+      using member_type = typename policy_type::member_type;
+      policy_type policy(AA.extent(0), Kokkos::AUTO(), AA.extent(5));
+      Kokkos::parallel_for
+        ("setTridiagToIdentity",
+         policy, KOKKOS_LAMBDA(const member_type &member) {
+          const int i = member.league_rank();
+          Kokkos::parallel_for(Kokkos::TeamThreadRange(member,AA.extent(1)),[&](const int &j) {
+              Kokkos::parallel_for(Kokkos::ThreadVectorRange(member, AA.extent(5)),[&](const int &v) {
+                  for (int k=0,kend=AA.extent(3);k<kend;++k)
+                    AA(i, j, 1, k, k, v) = 1;
+                });
+            });
+        });
+      Kokkos::fence();
+      const double t = timer.seconds();
+#if defined(KOKKOS_ENABLE_CUDA) && defined(KOKKOSBATCHED_PROFILE)
+      cudaProfilerStop();
+#endif
+      printf("identity time = %f\n", t);
+    }
+
+    /// randomize input
+    Kokkos::Random_XorShift64_Pool<exec_space> random(13245);
+    Kokkos::fill_random(As, random, value_type(1.0));
+    Kokkos::fill_random(bs, random, value_type(1.0));
+
+    Kokkos::deep_copy(Acopy, As);
+
+    ///
+    /// factorize the matrix
+    ///
+    if (1) {
```

Comment:
Why if(1) ?

Reply:
Just my convenience to test and profile the code separately. Profiler returns with weird concatenated names that I cannot figure out which kernels corresponds to which profile output. For convenience, I will keep this if statement.

Reply:
This a strange way to profile kernels, but I won't complain :)